generate-assets/src/bin/generate.rs

```diff
@@ -48,12 +48,67 @@ fn main() -> anyhow::Result<()> {
         },
     )?;
 
+    let latest_bevy_version = get_latest_bevy_version(&db)?;
+
+    sort_section(&mut asset_root_section.content, &latest_bevy_version);
+
     asset_root_section
         .write(Path::new(&content_dir), Path::new(""), 0)
         .expect("Failed to write assets section");
     Ok(())
 }
 
+/// Sort the assets in the section so that:
+/// - Assets that have been manually assigned an order in `bevy-assets` are first
+/// - Assets that are semver compatible with Bevy are next
+/// - If all else is equal, sort randomly
+fn sort_section(nodes: &mut [AssetNode], latest_bevy_version: &semver::Version) {
+    for node in nodes.iter_mut() {
+        if let AssetNode::Section(ref mut section) = node {
```

Comment:
```suggestion
    for node in nodes {
        if let AssetNode::Section(section) = node {
```
This seems to still compile.

Reply:
Are you sure? Does not compile on my end and I'm a bit confused about how it possibly could.

Reply:
Hm. This compiled on the Rust playground specifically:
```rust
struct Section {
    content: Vec<AssetNode>,
}

enum AssetNode {
    Section(Section),
}

fn sort_section(nodes: &mut [AssetNode]) {
    for node in nodes {
        if let AssetNode::Section(section) = node {
            sort_section(&mut section.content);
        }
    }
}
```

Reply:
I am doing more stuff than that in the real code though.

Reply:
Okay, so `IntoIter` on an `&mut [T]` will consume it apparently. This doesn't happen for `&[T]`...

Amended suggestion:

```suggestion
    for node in nodes.iter_mut() {
        if let AssetNode::Section(section) = node {
```

Reply:
Oh. It's because `&T` is `Copy`, but `&mut T` is not.